### PR TITLE
Made decodeStoredItem detect item type

### PR DESCRIPTION
**Pull request by OfficialNoob**

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -138,18 +138,19 @@ function encodeStoredItem(data) {
...
-        
-        if(!Array.isArray(str)) return data;
-        return new SBMap(id, str);
-    } catch(e) {
-
-        // If all else fails, return the data
-        return data;
+    if(!Config.localConfig[id]) return data;
+
+    if(Config.localConfig[id] instanceof SBMap) {
```
Comment (ajayyy) [inline src/config.ts]: Won't it be an array?

Comment (OfficialNoob) [reply src/config.ts]: Config.localConfig[id] should be the config value i think

Comment (ajayyy) [reply src/config.ts]: Yea, but chrome stores SBMap as an array right (with decode)

Comment (OfficialNoob) [reply src/config.ts]: Config.localConfig[id] is its old value

Comment (ajayyy) [reply src/config.ts]: This is decoding it. Did you mean to use `Config.defaults`?

Comment (OfficialNoob) [reply src/config.ts]: well it should be added when addDefaults() gets run.
however Config.defaults would probably be better to use

Comment (OfficialNoob) [reply src/config.ts]: Is this resolved?

**Comment by OfficialNoob** [comment]
@ajayyy its done.

**Comment by ajayyy** [comment]
@OfficialNoob You can now test the config with `window.SB.config`. Tell me how it goes.

File: src/config.ts [hunk 2]
```diff
@@ -138,18 +137,32 @@ function encodeStoredItem(data) {
...
+        try {
+            if (!Array.isArray(data)) return data;
+            return new SBMap(id, data);
+        } catch(e) {
+            console.error("Failed to parse SBMap: "+ id);
+        }
+    }
+
+    // This is the old format for SBMap (a JSON string)
+    if (typeof data === "string") {
```
Comment (OfficialNoob) [inline src/config.ts]: I was trying to prevent this.
I don't want a string being converted to a map.

Comment (ajayyy) [reply src/config.ts]: This is required to migrate the old data right?

Comment (OfficialNoob) [reply src/config.ts]: Well it should be migrated after the version change
not on every read however I don't think any map needs to be migrated

Comment (ajayyy) [reply src/config.ts]: It's easier to just migrate it this way. It can be removed in a few months.

Comment (OfficialNoob) [reply src/config.ts]: It would be better to have it during the update process.
at least it will be fixed some day :D

**Comment by ajayyy** [comment]
This has less chance of bugs occuring.

**Comment by OfficialNoob** [comment]
Yeah as long as there never a string like [] for any key

**Comment by OfficialNoob** [comment]
Tested current changes and it seems to work.
`SB.config` works now :D

**Comment by ajayyy** [comment]
Looks like that migration code I added is not working since it is not working if `sponsorTimes` when updating and having left over config data.

**Comment by ajayyy** [comment]
Seems good now I think

**Comment by ajayyy** [comment]
@OfficialNoob Can you check it?

**Comment by OfficialNoob** [comment]
@ajayyy tested it.

- Changes seem to result in the extension working :D
- SB.config seems to be no longer usable from the console.
- Enabling the skipper from the extension popup does not affect the current video without refresh. (Probably unrelated)

**Comment by ajayyy** [comment]
@OfficialNoob It should work on the console on the background page and options page.

**Comment by OfficialNoob** [comment]
Works fine at the options page

**Comment by ajayyy** [comment]
This is for security reasons. Before, any extension with access to the YouTube page could grab the user id

**Comment by OfficialNoob** [comment]
ok I think this is done now

**Comment by ajayyy** [comment]
I'll merge it once I figure out what is causing the freezing issue

**Pull request merged**